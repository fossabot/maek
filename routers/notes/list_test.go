package notes_test

import (
	"context"
	"fmt"
	"testing"

	approvals "github.com/customerio/go-approval-tests"
	"github.com/stretchr/testify/assert"

	"github.com/karngyan/maek/domains/notes"
	"github.com/karngyan/maek/zarf/tests"
)

func TestList(t *testing.T) {
	defer tests.CleanDBRows()

	cs := tests.NewClientStateWithUser(t)

	_, err := notes.UpsertNoteCtx(context.Background(), notes.WithUuid("123"), notes.WithContent("{ \"foo\": \"bar\" }"), notes.WithWorkspace(cs.Workspace), notes.WithUpdatedBy(cs.User), notes.WithFavorite(true))
	assert.Nil(t, err)

	rr, err := cs.Get(fmt.Sprintf("/v1/workspaces/%d/notes", cs.Workspace.Id))
	assert.Nil(t, err)
	assert.Equal(t, 200, rr.Code)

	approvals.VerifyJSONBytes(t, rr.Body.Bytes())
}
