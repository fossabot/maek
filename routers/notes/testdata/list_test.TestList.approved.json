{
  "nextCursor": "MQ==",
  "notes": [
    {
      "content": {
        "foo": "bar"
      },
      "created": 1234567890,
      "createdBy": {
        "created": 1234567890,
        "defaultWorkspaceId": 1,
        "email": "karn@maek.ai",
        "id": 1,
        "name": "Karn",
        "role": "admin",
        "updated": 1234567890,
        "verified": false
      },
      "favorite": true,
      "id": 1,
      "trashed": false,
      "updated": 1234567890,
      "updatedBy": {
        "created": 1234567890,
        "defaultWorkspaceId": 1,
        "email": "karn@maek.ai",
        "id": 1,
        "name": "Karn",
        "role": "admin",
        "updated": 1234567890,
        "verified": false
      },
      "uuid": "123",
      "workspaceId": 1
    }
  ]
}